ledger/participant-integration-api/src/test/lib/scala/platform/store/platform/partymanagement/ParticipantPartyStoreTests.scala

```diff
@@ -0,0 +1,281 @@
+                      Map(
+                        "k1" -> "v1b",
+                        "k3" -> "v3",
+                      )
+                    )
+                  ),
+                ),
+              ),
+              ledgerPartyExists = _ => Future.successful(true),
+            )
+            _ = update1.value shouldBe createdPartyRecord(
+              "party1",
+              resourceVersion = "1",
+              annotations = Map("k1" -> "v1b", "k2" -> "v2", "k3" -> "v3"),
+            )
+            // second update: with replace annotations semantics
+            update2 <- tested.updatePartyRecord(
+              partyRecordUpdate = PartyRecordUpdate(
+                party = pr.party,
+                metadataUpdate = ObjectMetaUpdate(
+                  resourceVersionO = None,
+                  annotationsUpdateO = Some(
+                    Replace(
+                      Map(
+                        "k1" -> "v1c",
+                        "k4" -> "v4",
+                      )
+                    )
+                  ),
+                ),
+              ),
+              ledgerPartyExists = _ => Future.successful(true),
+            )
+            _ = update2.value shouldBe createdPartyRecord(
+              "party1",
+              resourceVersion = "2",
+              annotations = Map("k1" -> "v1c", "k4" -> "v4"),
+            )
+            // third update: with replace annotations semantics - effectively deleting all annotations
+            update3 <- tested.updatePartyRecord(
+              partyRecordUpdate = PartyRecordUpdate(
+                party = pr.party,
+                metadataUpdate = ObjectMetaUpdate(
+                  resourceVersionO = None,
+                  annotationsUpdateO = Some(Replace(Map.empty)),
+                ),
+              ),
+              ledgerPartyExists = _ => Future.successful(true),
+            )
+            _ = update3.value shouldBe createdPartyRecord(
+              "party1",
+              resourceVersion = "3",
+              annotations = Map.empty,
+            )
+          } yield {
+            succeed
+          }
+        }
+      }
+
+      "should raise an error when updating a non-existing party record for which a ledger party doesn't exist" in {
+        testIt { tested =>
+          val party = Ref.Party.assertFromString("party")
+          for {
+            res1 <- tested.updatePartyRecord(
+              partyRecordUpdate = PartyRecordUpdate(
+                party = party,
+                metadataUpdate = ObjectMetaUpdate(
+                  resourceVersionO = None,
+                  annotationsUpdateO = Some(Merge.fromNonEmpty(Map("k1" -> "v1"))),
+                ),
+              ),
+              ledgerPartyExists = _ => Future.successful(false),
+            )
+            _ = res1.left.value shouldBe ParticipantPartyRecordStore.PartyNotFound(party)
+          } yield succeed
+        }
+      }
+    }
+  }
+
+}
```

Comment:
Missing test for incorrect resourceVersion in an update

Reply:
Added